### PR TITLE
Automatically configure webhooks after the completing the OAuth flow

### LINKED ISSUE #3292: Webhooks should be configured automatically when connecting via OAuth
Part of https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3196

When a store connects their Stripe account via the OAuth flow, we should automatically configure their webhooks too. 

In 8.4.0 we introduced a way for merchants to (re)configure webhooks automatically via a button (see https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3209). In 8.5 we're replacing the account key fields with an OAuth flow, as part of that, we should also make sure we configure webhooks automatically so the user doesn't need to click that button once they complete the OAuth connection.

### Requirements

- [ ] After arriving back on the site after the OAuth, configure webhooks.

### PR DESCRIPTION
**Pull request by james-allan**
Fixes #3292

## Changes proposed in this Pull Request:

After the user completes the OAuth Stripe flow, we now configure the webhooks automatically. This will make it a lot simpler for merchants and make it far more likely that sites will have their webhooks configured correctly. 

## Testing instructions

1. Go to **WooCommerce > Settings > Payments > Stripe (settings).**
2. Open the Account keys/connection modal.
3. Go to the the **Test** tab.
   - The test tab wont work until we are able to connect to test accounts via the OAuth flow - https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3291. 
4. Click the **Connect and account** button. 
6. You will be redirected to the OAuth page hosted by Stripe.
7. Login and select an account.
8. Click **Connect**.
9. At this point you can abandon the rest of the set up and just click the "Return to WooCommerce Inc." on the left.
10. Once you're back on the site, reopen the account keys modal.
11. You should see that the webhooks have been configured - the Webhooks status will configured and the webhook URL will appear at the bottom.
12. Check your webhooks in the Stripe Dashboard and confirm the webhook was created.

----

<p align="center">
Screenshot 2024-07-19 at 4 30 32 PM
Screenshot 2024-08-02 at 3 15 51 PM

<br>
<sup>Webhooks configured after OAuth flow</sup>
</p> 

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: includes/connect/class-wc-stripe-connect.php [hunk 1]
```diff
@@ -154,6 +156,9 @@ private function save_stripe_keys( $result, $mode = 'live' ) {
 
 			update_option( self::SETTINGS_OPTION, $options );
 
+			// Automatically configure webhooks for the account now that we have the keys.
+			WC_Stripe::get_instance()->account->configure_webhooks( $is_test ? 'test' : 'live', $secret_key );
```
Comment (Mayisha) [inline includes/connect/class-wc-stripe-connect.php]: The exception thrown in the `configure_webhooks` is not handled here. For testing purpose, I intentionally created an error condition [in the function](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3293/files#diff-4314dc04e3c11401b896736a579e7b2cce4634bab67b790ef2872cc6b39edfb3R300) and got this fatal error.
 
Screenshot 2024-08-02 at 5 43 40 PM

**Review by Mayisha — CHANGES_REQUESTED** [review]
Thanks @james-allan for implementing this important feature.

The webhook configuration is working as described. When I connect my account, it configures the webhook properly.

Screenshot 2024-08-02 at 5 35 35 PM

Under the newly configured webhook in the Stripe dashboard it has added 17 events to listen to.

Screenshot 2024-08-02 at 6 44 07 PM

I tried to simulate an error situation and ended up getting a fatal error which I mentioned in my comment. We should handle this one.

Apart from that, after completing the connection flow, I see this webhook related warning on my Stripe settings page. This one might be unrelated to the PR though.

Screenshot 2024-08-02 at 5 35 12 PM

File: includes/connect/class-wc-stripe-connect.php (hunk 1 above)
Comment (james-allan) [reply includes/connect/class-wc-stripe-connect.php]: Good call. I've fixed this in [e74c110](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3293/commits/e74c110d1c31b8c975f64ef9194f4b28e6b24add) by catching an exception and returning a `WP_Error` which is consistant with this function returning error objects.

**Comment by james-allan** [comment]
> Apart from that, after completing the connection flow, I see this webhook related warning on my Stripe settings page. This one might be unrelated to the PR though.

I'm not 100% sure what causes that. I think what's happening is your store is receiving webhooks for multiple endpoints but it's failing the validation on the stored webhook secret and therefore leading to that error being shown. There's an open issue for clearing the state after configuring a webhook: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3280. I believe that should address that.

**Review by Mayisha — APPROVED** [review]
Thanks @james-allan for fixing the error case. I can not reproduce it anymore and also the webhook configuration is working perfectly on completing the OAuth flow 🎉

**Pull request merged**